### PR TITLE
Download subtitles of a YouTube video (#9) 

### LINKED ISSUE #9: Download subtitles of a YouTube video
Download the subtitles of a YouTube video as `video.srt`

To achieve this, use `youtube-dl` with the `--write-auto-sub` option
The solution should go at [Line 124](https://github.com/OpenGenus/vidsum/blob/master/code/sum.py#L124)

Other solutions are warmly welcomed as well.

### PR DESCRIPTION
**Pull request by codebu5ter**
Closes #9

### CODE REVIEW

File: code/sum.py [hunk 1]
```diff
@@ -108,20 +109,42 @@ def download_video(url):
...
+    '''
+    ydl_opts = {
+            'outtmpl': '%(title)s.%(ext)s',
+            'subtitlesformat':'srt',
+            'writeautomaticsub':True,
+            'skip_download':True
+            }
+
+    with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+        ydl.download([subs])
```
Comment (AdiChat) [inline code/sum.py]: What will be the filename of the subtitle file?

Ideally, I should be consistent such as `1.srt`

File: code/sum.py [hunk 2]
```diff
@@ -108,20 +109,42 @@ def download_video(url):
...
     args = parser.parse_args()
 
     url = args.url
-    if not url:
-        get_summary(args.video_file, args.subtitles_file)
-    else:
-        download_video(url)
+    subs = args.subs
+
+    if subs:
```
Comment (AdiChat) [inline code/sum.py]: Ideally, if video url is provided, the video and subtitles should be downloaded together as the final objective is to summarize a YouTube video provided its url.

Kindly make the corresponding changes.

File: code/sum.py (hunk 1 above)
Comment (codebu5ter) [reply code/sum.py]: @AdiChat I've changed the filename to 1.srt, as requested.

File: code/sum.py (hunk 2 above)
Comment (codebu5ter) [reply code/sum.py]: @AdiChat I've now modified my function to download the video along with its subtitles. It now performs both downloads in a single function and doesn't require the pytube module anymore.

**Comment by codebu5ter** [comment]
@AdiChat Do you want me to squash all my commits into a single commit?

File: code/sum.py [hunk 3]
```diff
@@ -95,17 +94,22 @@ def get_summary(filename="1.mp4", subtitles="1.srt"):
...
-    video = yt.filter('mp4')[-1]
-    video.download(os.getcwd())
+    ydl_opts = {
+            'outtmpl': '1.%(ext)s',
+            'subtitlesformat':'srt',
+            'writeautomaticsub':True
+            }
+
+    with youtube_dl.YoutubeDL(ydl_opts) as ydl:
+        ydl.download([subs])
```
Comment (AdiChat) [inline code/sum.py]: Add comments for the following:
What will be the filename and location of the downloaded video and subtitle? Will it ensure the highest resolution as resolved in a previous pull request?

These details are required for further development.

File: code/sum.py [hunk 4]
```diff
@@ -95,17 +94,22 @@ def get_summary(filename="1.mp4", subtitles="1.srt"):
     summary.to_videofile( output, codec="libx264", temp_audiofile="temp.m4a", remove_temp=True, audio_codec="aac")
     return True
 
-def download_video(url):
-    ''' Downloads specified Youtube video
+# download video with subtitles
+def downvdo_subs(subs):
```
Comment (AdiChat) [inline code/sum.py]: Kindly rename as `def download_video_srt(url):`

Also, make corresponding changes 👍

**Comment by AdiChat** [comment]
Make the proposed changes and it is good to be merged. No need to squash the commits.

File: code/sum.py (hunk 4 above)
Comment (codebu5ter) [reply code/sum.py]: @AdiChat Consider it done 👍

File: code/sum.py (hunk 3 above)
Comment (codebu5ter) [reply code/sum.py]: @AdiChat added the comments, as requested.

**Comment by codebu5ter** [comment]
@AdiChat Done! :)

**Comment by AdiChat** [comment]
Looks good 👍 

Hope you are enjoying your journey with **OpenGenus**

**Comment by AdiChat** [comment]
@codebu5ter Add yourself to the [Contributor list](https://github.com/OpenGenus/vidsum/blob/master/contributors.md) 🎉 in a pull request.

**Comment by codebu5ter** [comment]
@AdiChat I'm on it. :)

**Pull request merged**